core/src/main/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcher.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.applicationinsights.internal.jmx;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+
+/**
+ * A utility class that knows how to fetch JMX data.
+ *
+ * Created by gupele on 3/15/2015.
+ */
+public class JmxDataFetcher {
+    /**
+     * Gets an object name and its attributes to fetch and will return the data.
+     * @param objectName The object name to search.
+     * @param attributes The attributes that 'belong' to the object name.
+     * @return A map that represent each attribute: the key is the displayed name for that attribute
+     * and the value is a list of values found
+     * @throws Exception In case the object name is not found.
+     */
+    public static Map<String, Collection<Object>> fetch(String objectName, Collection<JmxAttributeData> attributes) throws Exception {
```

Comment:
The reason it throws is that if there is a major problem like the object name is not ok, then there is no point of trying again